packages/store/src/internal/dispatcher.ts

```diff
@@ -46,15 +44,15 @@ export class InternalDispatcher {
     return result.pipe(leaveNgxs(this._ngxsExecutionStrategy));
   }
 
-  private dispatchByEvents(actionOrActions: any | any[]): Observable<any> {
-    if (Array.isArray(actionOrActions)) {
-      return forkJoin(actionOrActions.map(a => this.dispatchSingle(a)));
+  private dispatchByEvents<T = any>(actions: ActionType | ActionType[]): Observable<any> {
```

Comment:
**_Stopping my review at this point. There are just too many changes in this review to be able to know that we are not breaking something._**